packages/relay-kit/src/packs/safe-4337/Safe4337Pack.ts

```diff
@@ -476,9 +520,15 @@ export class Safe4337Pack extends RelayKitBasePack<{
    * @param {EthSafeOperation} safeOperation - The SafeOperation to execute.
    * @return {Promise<string>} The user operation hash.
    */
-  async executeTransaction({
-    executable: safeOperation
-  }: Safe4337ExecutableProps): Promise<string> {
+  async executeTransaction({ executable }: Safe4337ExecutableProps): Promise<string> {
+    let safeOperation: EthSafeOperation
+
+    if ('safeOperationHash' in executable) {
```

Comment:
How about defining this as a type guard, so you could use it as `!isSafeOperation(executable)` here, as well as in line 467. You could even use it in [SafeApiKit line 795](https://github.com/safe-global/safe-core-sdk/blob/feat/add-api-compatibility-to-4337-pack/packages/api-kit/src/SafeApiKit.ts#L795)

Reply:
Hey, changed. Defined the guards in the types library together with the type definition